packages/docusaurus/src/client/clientEntry.tsx

```diff
@@ -30,18 +35,43 @@ if (ExecutionEnvironment.canUseDOM) {
   // first-load experience.
   // For development, there is no existing markup so we had to render it.
   // We also preload async component to avoid first-load loading screen.
-  const renderMethod =
-    process.env.NODE_ENV === 'production' ? ReactDOM.hydrate : ReactDOM.render;
-  preload(routes, window.location.pathname).then(() => {
-    renderMethod(
-      <HelmetProvider>
-        <BrowserRouter>
-          <App />
-        </BrowserRouter>
-      </HelmetProvider>,
+  if (!isReactRoot()) {
+    // for react <= 17
```

Comment:
I don't really like this. We should just release this as a breaking change and ask everyone to migrate to React v18.